tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -1002,6 +1004,158 @@ TYPED_TEST(
+{
+	const auto m = 2;
+	SGMatrix<TypeParam> A(m, m);
+	SGMatrix<TypeParam> B(m, m);
+	SGMatrix<TypeParam> result(m, m);
+
+	std::iota(A.matrix, A.matrix + A.size(), 0);
+	std::iota(B.matrix, B.matrix + B.size(), 1);
+	
+	element_div(A, B, result);
+
+	for (const auto& [result, el1, el2]: zip_iterator(result ,B , A))
+    		EXPECT_NEAR(el2 / el1, result, get_epsilon<TypeParam>());
+
+	SGMatrix<TypeParam> C(m+1, m);
+	SGMatrix<TypeParam> D(m, m+1);
+	SGMatrix<TypeParam> E(m+1, m+1);
+
+	EXPECT_THROW(element_div(A, B, C), ShogunException);
+	EXPECT_THROW(element_div(A, B, D), ShogunException);
+	EXPECT_THROW(element_div(A, B, E), ShogunException);
+
+	EXPECT_THROW(element_div(A, C, B), ShogunException);
+	EXPECT_THROW(element_div(A, D, B), ShogunException);
+	EXPECT_THROW(element_div(A, E, B), ShogunException);
+
+	EXPECT_THROW(element_div(C, A, B), ShogunException);
+	EXPECT_THROW(element_div(D, A, B), ShogunException);
+	EXPECT_THROW(element_div(E, A, B), ShogunException);
+
+	if constexpr (std::is_floating_point<TypeParam>::value){
+	B.zero();
+	element_div(A, B, result);
+	typename SGMatrix<TypeParam>::EigenMatrixXtMap result_eig = result;
+    	EXPECT_TRUE(result_eig.array().isInf().any() && result_eig.array().isNaN().any());
+	}
+}
+
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGMatrix_block_elementwise_division)
+{
+	const index_t nrows = 2;
+	const index_t ncols = 3;
+
+	SGMatrix<TypeParam> A(nrows, ncols);
+	SGMatrix<TypeParam> B(ncols, nrows);
+
+	for (auto i : range(nrows))
+		for (auto j : range(ncols))
+		{
+			A(i, j) = i * 10 + j + 1;
+			B(j, i) = i + j + 1;
+		}
+
+	const auto m = 2;
+	auto A_block = linalg::block(A, 0, 0, m, m);
+	auto B_block = linalg::block(B, 0, 0, m, m);
+	auto result = element_div(A_block, B_block);
+
+	ASSERT_EQ(result.num_rows, m);
+	ASSERT_EQ(result.num_cols, m);
+
+	for (auto i : range(m))
+		for (auto j : range(m))
+			EXPECT_NEAR(
+			    result(i, j), A(i, j) / B(i, j), get_epsilon<TypeParam>());
+
+	auto C_block = linalg::block(B, 0, 0, m, m+1);
+	EXPECT_THROW(element_div(A_block, C_block),ShogunException);
+	C_block = linalg::block(B, 0, 0, m+1, m);
+	EXPECT_THROW(element_div(A_block, C_block),ShogunException);
+	C_block = linalg::block(B, m-1, 0, m, m+1);
+	EXPECT_THROW(element_div(A_block, C_block),ShogunException);
+
+}
+
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGVector_elementwise_division)
+{
+	const index_t len = 4;
+	SGVector<TypeParam> a(len);
+	SGVector<TypeParam> b(len);
+	SGVector<TypeParam> c(len);
+
+	a.range_fill();
+	b.range_fill(1);
+
+	c = element_div(a, b);
+
+	for (const auto& [result, el1, el2]: zip_iterator(c,b,a))
+    		EXPECT_NEAR(el2 / el1, result, get_epsilon<TypeParam>());
+
+	if constexpr (std::is_floating_point<TypeParam>::value){
+	b.zero();
+	c = element_div(a, b);
+	typename SGVector<TypeParam>::EigenRowVectorXtMap result_eig = c;
+    	EXPECT_TRUE(result_eig.array().isInf().any() && result_eig.array().isNaN().any());
+	}
```

Comment:
indent

Reply:
just this missing

Reply:
Do I remove block operation as @karlnapf suggested?

Reply:
yes let's do that